spec/requests/pingometer_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'rails_helper'
+
+RSpec.describe 'Pingometer Webhook', type: :request do
+  it 'is persisted appropriately in the database' do
+    monitor_id = 'b74014410cc1236a3d0h7400'
+
+    # Expect a pre-existing monitor
+    web_service = WebService.create pingometer_id: monitor_id,
+      raw_monitor_data: {
+        'hostname': 'test.com'
+      }
+
+    # Expect a screenshot
+    expect(Browserstack).to receive(:screenshot) {
+      File.new(Rails.root.join('spec', 'fixtures', 'screenshot.png'), 'r')
+    }.at_least(:once)
+
+    pingometer_data = {
+      monitor_id: monitor_id,
+      monitor_host: 'pingometer.com',
+      monitor_status: 'down',
+      utc_timestamp: '2015-03-10T13:13:19' # Pingometer uses ISO8601 but without the Z
+    }
+    post '/hooks/event', pingometer_data.to_json, 'Content-Type' => 'application/json'
```

Comment:
The only data you get in the POST is `monitor_id`, `monitor_host` (unreliable; it’ll be blank for half the monitors because they are scripts and not simple HTTP requests; to get those hostnames requires getting metadata about the monitor from Pingometer), and `monitor_status` (new! This didn’t exist when I first wrote the hook). http://support.pingometer.com/knowledge_base/topics/what-is-the-webhook